torch/fx/graph.py

```diff
@@ -898,6 +898,21 @@ def generate_output(self, output_args):
             return super().generate_output(output_args)
 
 
+class _CompiledAutogradCodeGen(CodeGen):
+    def __init__(self, boxed_inputs_count):
+        super().__init__()
+        self._boxed_inputs_count = boxed_inputs_count
+
+    def process_inputs(self, *args):
+        # Compiled Autograd's runtime inputs contain activations
+        # that must be freed ASAP during execution. We do this
+        # by passing the args in a list that we "steal" at runtime.
+        # However, we still want compile-time analysis to be done
+        # on unpacked inputs as we don't have first class support
+        # for lists. Hence, we unflatten the inputs here.
+        return (args[: self._boxed_inputs_count], *args[self._boxed_inputs_count :])
```

Comment:
Why aren't all inputs just boxed, wouldn't that be simpler?

Reply:
it's easier to handle a list of only tensors in dynamo variables, the rest of the inputs are symints, python callables

Reply:
More Q for my information: do you end up with an actual ListVariable in Dynamo when you trace the boxed input?

Reply:
yep, but we don't use the ListVariable after we unpack it into TensorVariables